metals/src/main/scala/scala/meta/internal/metals/codeactions/FlatMapToForComprehensionCodeAction.scala

```diff
@@ -246,29 +248,66 @@ class FlatMapToForComprehensionCodeAction(
 
     val newName = nameGenerator.createNewName()
     val newTerm = replacePlaceHolder(term, Term.Name(newName), true)
-    if (replacementTimes == 1) Some((newName, newTerm))
+    if (replacementTimes == 1) Some((Pat.Var(Term.Name(newName)), newTerm))
     else None
   }
 
-  private def processValueNameAndNextQual(
+  private def isSimple(pat: Pat): Boolean = { // this is to decide whether to
+    // put pat in the left side of an Enumerator
+    pat match { // TODO help required @Vadim
+      case _: Pat.Alternative => false
+      case _: Pat.Bind => false
+      case _: Pat.Extract => true
+      case _: Pat.ExtractInfix => true
+      case _: Pat.Given => false
+      case _: Pat.Interpolate => true
+      case _: Lit => true
+      case _: Pat.Macro => false
+      case _: Term.Name => true
+      case _: Pat.Quasi => false
+      case _: Pat.Repeated => false
+      case _: Term.Select => false
+      case _: Pat.SeqWildcard => false
+      case Pat.Tuple(pats) => pats.forall(isSimple)
+      case _: Pat.Typed => true
+      case _: Pat.Var => true
+      case _: Pat.Wildcard => false
+      case Pat.Xml(_, args) => args.forall(isSimple)
+      case _ => false
+    }
```

Comment:
There is no need to explicitly cover all possible pats(for example xml) - only these that occurs in tests.
I reduced it locally to and tests pass:
```scala
    pat match {
      case Pat.Tuple(pats) => pats.forall(isSimple)
      case _: Pat.Var => true
      case _: Term.Name => true
      case _: Pat.Typed => true
      case _ => false
    }
```
I'm not sure if all that I left is needed. Could you double check that?

Reply:
Hi @dos65 Thank you so much for the review. I have not included the extra cases which are covered in the tests, but they were working on scastie. I mean like XML. Do I add them to the tests?

For [example](https://scastie.scala-lang.org/vzmerr/tPvC34B4RM2EIywhJlcLRg/10) :
```
val a = <p> here </p>
val b = <p> over there </p>

val c = for {
  <p>{value}</p> <- List(a, b)
} yield value

println(c)
```

prints `List( here ,  over there )`

Reply:
oh! nice.
So, then I think it would better to cover the cases that are valid and supported by tests.

And anyway these blocks requires some refactoring. Put all valid option at the top and turn all options that returns false   into a single last wildcard: `case _ => false`